web/src/engine/predictive-text/worker-thread/src/main/correction/context-state.ts

```diff
@@ -86,4 +107,48 @@ export class ContextState {
       this.suggestions = [].concat(stateToClone.suggestions);
     }
   }
+
+  /**
+   * The context window in view for the represented Context state,
+   * as passed between the predictive-text worker and its host.
+   */
+  get context(): Context {
+    return this._context;
+  }
+
+  /**
+   * Initializes the ContextState instance for use when no valid prior
+   * information is available - typically, immediately after engine
+   * initialization or a context reset.
+   */
+  private initFromReset() {
+    const context = this.context;
+    const lexicalModel = this.model;
+
+    const tokenizedContext = determineModelTokenizer(lexicalModel)(context).left;
+
+    let baseTokens = tokenizedContext.map(function(entry) {
+      let token = new ContextToken(lexicalModel, entry.text);
+
+      if(entry.isWhitespace) {
+        token.isWhitespace = true;
+      }
+
+      return token;
+    });
+
+    // And now build the final context state object, which includes whitespace 'tokens'.);
+    const tokenization: ContextToken[] = [];
+
+    while(baseTokens.length > 0) {
+      tokenization.push(baseTokens.splice(0, 1)[0]);
+    }
+
+    if(tokenization.length == 0) {
+      let token = new ContextToken(lexicalModel);
+      tokenization.push(token);
+    }
+
+    this.tokenization = new ContextTokenization(tokenization);
```

Comment:
```suggestion
    if(baseTokens.length == 0) {
      baseTokens.push(new ContextToken(this.model));
    }

    this.tokenization = new ContextTokenization(baseTokens);
```

As far as I can tell, that's equivalent?